framework/src/tinyalsa/tinyalsa.c

```diff
@@ -1091,9 +1173,6 @@ int pcm_start(struct pcm *pcm)
 		return oops(pcm, errno, "cannot start channel");
 	}
 
-	pcm->nextSize = 0;
-	pcm->nextOffset = 0;
-	pcm->nextBuf = NULL;
 	pcm->running = 1;
```

Comment:
I feel it is better we set pcm->draining = 0 here .
consider the case :- 
pcm_open -> pcm_write ->pcm_drain -> pcm_start -> pcm_write call sequence.

Reply:
True. I will change it. 
Though, pcm->draining is not applicable in writei, the problem you have mentioned can happen in readi.